Wrappers/Python/ccpi/optimisation/operators/FiniteDifferenceOperator.py

```diff
@@ -23,25 +23,29 @@ class FiniteDiff(Operator):
     # Grad_order = ['direction_z', 'direction_y', 'direction_x']
     # Grad_order = ['channels', 'direction_z', 'direction_y', 'direction_x']
     
-    def __init__(self, gm_domain, gm_range=None, direction=0, bnd_cond = 'Neumann'):
+    def __init__(self, gm_domain, gm_range=None, direction=0, bnd_cond = 'Neumann', **kwargs):
         ''''''
         super(FiniteDiff, self).__init__() 
-        '''FIXME: domain and range should be geometries'''
+
         self.gm_domain = gm_domain
         self.gm_range = gm_range
+        
         self.direction = direction
         self.bnd_cond = bnd_cond
-        
+                
+        ''' Domain Geometry = Range Geometry'''
         if self.gm_range is None:
             self.gm_range = self.gm_domain
-            
-        if self.direction + 1 > len(gm_domain):
-            raise ValueError('Gradient directions more than geometry domain')      
-                         
+        
+        ''' check direction and "length" of geometry  '''
+        if self.direction + 1 > len(self.gm_domain):
```

Comment:
Is `len(self.gm_domain)` defined?

Geometries are not [`Collections` or `Sequences`](https://docs.python.org/3/library/functions.html#len) so the `len` builtin should not work in this case.  Maybe this is what you want?
```python
len(self.gm_domain.shape)
```